graph/pkg/service/v0/drives.go

```diff
@@ -104,15 +138,25 @@ func (g Graph) GetDrives(w http.ResponseWriter, r *http.Request) {
 		errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, err.Error())
 		return
 	}
-	files, err := g.formatDrives(ctx, wdu, res.StorageSpaces)
+	spaces, err := g.formatDrives(ctx, wdu, res.StorageSpaces)
 	if err != nil {
-		g.logger.Error().Err(err).Msg("error encoding response as json")
+		g.logger.Error().Err(err).Msg("error encoding response")
 		errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, err.Error())
 		return
 	}
-
-	render.Status(r, http.StatusOK)
-	render.JSON(w, r, &listResponse{Value: files})
+	switch num := len(spaces); {
+	case num == 0:
+		g.logger.Error().Str("driveId", driveId).Msg("no space found")
+		errorcode.ItemNotFound.Render(w, r, http.StatusNotFound, fmt.Sprintf("no space found with id %s", driveId))
+		return
+	case num == 1:
+		render.Status(r, http.StatusOK)
+		render.JSON(w, r, spaces[0])
+	default:
+		g.logger.Error().Int("number", num).Msg("an unexpected number of spaces was found")
```

Comment:
this error message is really vague, it tells nothing to the user of what happened or how to interpret it 🤔

Reply:
This is no user facing error. It is just a log message. How would you phrase it? It is pretty unlikely that this will happen at all.